api/krusty/remoteload_test.go

```diff
@@ -97,11 +101,45 @@ func runResourceTests(t *testing.T, cases map[string]*remoteResourceCase) {
 			if savedTest.local && !isLocalEnv(req) {
 				t.SkipNow()
 			}
+			configureGitSSHCommand(t)
 			testRemoteResource(req, test)
 		})
 	}
 }
 
+func configureGitSSHCommand(t *testing.T) {
+	t.Helper()
+
+	// This contains a read-only Deploy Key for the kustomize repo.
+	node, err := yaml.ReadFile("testdata/repo_read_only_ssh_key.yaml")
+	if err != nil {
+		t.Fatal("Failed to load ssh key: ", err)
+	}
+	keyB64, err := node.GetString("key")
+	if err != nil {
+		t.Fatal("Failed to load ssh key: ", err)
+	}
+	key, err := base64.StdEncoding.DecodeString(keyB64)
+	if err != nil {
+		t.Fatal("Failed to decode ssh key: ", err)
+	}
+
+	// Write the key to a temp file and use it in SSH
+	f, err := os.CreateTemp("", "kustomize_ssh")
+	if err != nil {
+		t.Fatal("Failed to write ssh key: ", err)
+	}
+	if _, err = io.Copy(f, bytes.NewReader(key)); err != nil {
+		t.Fatal("Failed to write ssh key: ", err)
+	}
+	cmd := fmt.Sprintf("ssh -i %s", f.Name())
+	const SSHCommandKey = "GIT_SSH_COMMAND"
+	t.Setenv(SSHCommandKey, cmd)
```

Comment:
I believe this test suite runs in parallel (most of our tests do), which means we could have a race here: one test could be unsetting the env between when another sets and uses it. Ideally, we would set this once at the beginning of a test run.